### DISCUSSION TITLE
🚀 New DEV Branch with Critical Updates is Live for Testing! 🛠️

### DISCUSSION THREAD

**Discussion by MickLesk** [community-scripts/ProxmoxVE#715] [Announcements]
Dear Community,  

We’re thrilled to announce that a new **DEV Branch** is now available! 🎉 This branch incorporates **all changes** from two critical pull requests:  

- **[PR #643: Massive Core Dependencies Update](https://github.com/community-scripts/ProxmoxVE/pull/643)**  
- **[PR #710: Standardization and Improvements](https://github.com/community-scripts/ProxmoxVE/pull/710)**  

Before diving into testing, make sure to **backup your system** 🛡️. While we’ve tested the changes thoroughly, this branch is still in development, and issues may occur.  

---

## 🔧 How to Test the New Scripts  
You can test all scripts from the website with a small modification:  

1️⃣ Replace `main` with `develop` in the download URL.  
   
Example for the **Vikunja-LXC** script:  
```  
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/vikunja.sh)"
```  
becomes:  
```  
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/develop/ct/vikunja.sh)"
```  

Feel free to share any issues directly in this post! 🚨  But check first, if this issue is recent in main branch too.

---

### 🔍 PR #643: Massive Core Dependencies Update  
This is one of the largest updates in a long time! Highlights include:  

- **New Features**  
  - Added **tagging support** for `CT.sh` files, tags now transfer to Proxmox.  
  - Enhanced **MOTD** for all Systems
  - Optimized **Base Default Settings** in `CT.sh` with better customization options.  
  - Improved **ping checks** for IPv4 and IPv6.  
  - Revamped UI/UX: consistent color-coding, updated spinners, and clearer layout.  
  - **Alpine installation improvements** with the latest stable 3.20 support.  
  - Streamlined `CT.sh` functionality while maintaining backward compatibility.  

- **Bug Fixes**  
  - Fixed and shortened `exit_script`.  
  - Improved error handling and variable declarations.  
  - Adjusted outputs for advanced settings to align with default design.  

- **Other Changes**  
  - Removed outdated RPM script areas from `update_os`.  
  - Further optimizations in indentation, formatting, and naming conventions.

### 🛠️ PR #710: Standardization and Improvements  
- Standardized script headers and completions.  
- Merged license and author information for cleaner metadata.  
- Unified default values with privileged differentiation.  
- Added **source URLs** for better traceability.  
- Consistent indentation and formatting (Visual Studio Code compatible).  

---

## 📝 Feedback and Reporting Issues  
We encourage you to test the scripts and share any feedback or issues directly in this post. Your contributions are invaluable!  

Happy testing, and don’t forget to **backup your system** before diving in! 😄

**Comment by dsiebel** [comment]
Not an issue per se, but I find the amount of emojis used borderline disturbing 😅 
It distracts from the actual content, and, in my opinion, doesn't look nice at all.
Less is more!

![image](https://github.com/user-attachments/assets/088d7ca1-37c3-4936-9814-d0875c6b56b6)

![image](https://github.com/user-attachments/assets/511ae507-eaf9-4127-a032-621345018a3d)

**Reply by MickLesk** [reply]
Only in Advanced. In next Version output of Changed advanced Settings. It will be less Output

**Comment by michelroegl-brunner** [comment]
Hey @MickLesk!

I´ve tested the new system with a few scripts so far and did not run in any problems. Great work you did there!

Just one small thing which is catching my eye every time i install a new container, altough it´s a purely asthetic one! 
![Circle](https://github.com/user-attachments/assets/12c9e7bf-da16-4d10-909f-f6cd62d67dd6)

Can you maybe put some spacing between the spinner and the text? For me personaly, i find it more opticly pleasing with one space or tab in between.

**Reply by MickLesk** [reply]
I saw it, I mistakenly removed the space when I fixed the indentations ^^

Will be fixed.

**Reply by michelroegl-brunner** [reply]
Perfekt^^

And aren´t the tags defined in the /ct/ script supposed to be transfered to proxmox? 
I´ve just seen it not working with snipeit.sh, there are no tags beside the comunity-script one.

![snipeit](https://github.com/user-attachments/assets/659ba322-4c38-446f-9246-dfb8f3a3784d)

**Reply by MickLesk** [reply]
Default or Advanced?

**Reply by michelroegl-brunner** [reply]
> Default or Advanced?

Default

**Reply by michelroegl-brunner** [reply]
It is also not working in advanced mode. And my host is running pve-manager/8.3.1/fb48e850ef9dde2.

**Reply by MickLesk** [reply]
I Check it tomorrow

**Reply by MickLesk** [reply]
both fixed

**Comment by bvdberg01** [comment]
I tested it with a few scripts, but I have a strange problem with the NetBox script.

When I install with dev branch I get default Apache page:

![Screenshot 2024-12-07 001351](https://github.com/user-attachments/assets/450bc89b-9c4f-4ed6-a973-3e2b7e4eb1fe)
![Screenshot 2024-12-07 001518](https://github.com/user-attachments/assets/4a11054c-d075-4b4b-8cc5-42d8bfd6d920)


When I install with main branch I have a working NetBox:
![Screenshot 2024-12-07 002542](https://github.com/user-attachments/assets/7db25c30-2faa-41d2-b85b-66a9b9daeaea)
![Screenshot 2024-12-07 002644](https://github.com/user-attachments/assets/29ed19fe-3ad2-458e-af2d-a66713e53ea7)

 I know how to solve the issue (add a2dissite 000-default), but maybe more scripts have the same issue?

**Reply by MickLesk** [reply]
HTTPS vs http. Wrong in the Script

**Reply by MickLesk** [reply]
Fixed

**Comment by michelroegl-brunner** [comment]
I´ve got some sort of "feature" request if it is possible.

When installing in verbose mode the spinner is also rendered. I know it is needed to indicate the script is still running while not having any outuput. but in verbose mode, in my opinion. it is not really needed and more a distraction and it looks a bit off.

![spinner](https://github.com/user-attachments/assets/838c7a33-3927-4767-9c62-94b18ea0f68e)
 
Prob. you could think about suppressing the spinner when in verbose mode :)

**Reply by MickLesk** [reply]
Not so easy, this is inherited from the main function. in the old version it is present to. :D 

![image](https://github.com/user-attachments/assets/33993fc0-c0d3-41c1-bba0-0383ec0d7fc0)

**Reply by michelroegl-brunner** [reply]
You could do something like this in _install.func_ :

```
spinner() {
   if [ $VERBOSE != "yes" ]; then
    local frames=('⠋' '⠙' '⠹' '⠸' '⠼' '⠴' '⠦' '⠧' '⠇' '⠏')
    local spin_i=0
    local interval=0.1
  
    printf "\e[?25l"
     local color="${YWB}"
     while true; do
      printf "\r ${color}%s${CL}" "${frames[spin_i]}"
      spin_i=$(( (spin_i + 1) % ${#frames[@]} ))
      sleep "$interval"
    done
  fi
 }
```
Then the spinner will not get startet^^

**Reply by MickLesk** [reply]
but there is no spinner for every msg_info, that looks very blank before the cross or checkmark set ^^
